### ISSUE TITLE
nvidia-driver-daemonset pod CrashLoopBackOff after restart

### ISSUE THREAD

**Issue by rbo** [NVIDIA/gpu-operator#83]
### 1. Quick Debug Checklist
- [ ] Are you running on an Ubuntu 18.04 node?
- [ ] Are you running Kubernetes v1.13+?
- [ ] Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?
- [ ] Do you have `i2c_core` and `ipmi_msghandler` loaded on the nodes?
- [x] Did you apply the CRD (`kubectl describe clusterpolicies --all-namespaces`)
```
$ kubectl describe clusterpolicies --all-namespaces
Name:         cluster-policy
Namespace:
Labels:       <none>
Annotations:  <none>
API Version:  nvidia.com/v1
Kind:         ClusterPolicy
Metadata:
  Creation Timestamp:  2020-09-20T12:31:58Z
  Generation:          1
  Resource Version:    511518
  Self Link:           /apis/nvidia.com/v1/clusterpolicies/cluster-policy
  UID:                 25d885e7-63b5-4f1c-b0c5-d0dd4f1cc52e
Spec:
  Dcgm Exporter:
    Image:       dcgm-exporter
    Repository:  nvidia
    Version:     1.7.2-2.0.0-rc.9-ubi8
  Device Plugin:
    Image:       k8s-device-plugin
    Repository:  nvidia
    Version:     1.0.0-beta6-ubi8
  Driver:
    Image:       driver
    Repository:  nvidia
    Version:     440.64.00
  Operator:
    Default Runtime:  crio
  Toolkit:
    Image:       container-toolkit
    Repository:  nvidia
    Version:     1.0.2-ubi8
Status:
  State:  ready
Events:   <none>
$
```
### 1. Issue or feature description

If you delete the nvidia-driver-daemonset pod after a successful NVIDIA GPU Operator 1.1.7-r2 installation on OpenShift 4.4.16 the new pod stucks in  CrashLoopBackOff. The problem is 
```
+ echo 'Could not unload NVIDIA driver kernel modules, driver is in use'
Could not unload NVIDIA driver kernel modules, driver is in use
```
[nvidia-driver-daemonset-phpf7-nvidia-driver-ctr.log](https://github.com/NVIDIA/gpu-operator/files/5251877/nvidia-driver-daemonset-phpf7-nvidia-driver-ctr.log)

Of course, the kernel module is still loaded. Rebooting the node solves the problem because the daemon set an install and load the kernel again.  If you delete the pod again, the problem still exists. 

Technically it is not really a problem, GPUs work well any time. But the user experience not very handy. 

Actions

### 2. Steps to reproduce the issue

Delete one if the nvidia-driver-daemonset pod's.

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

 - [ ] kubernetes pods status: `kubectl get pods --all-namespaces`
 - [ ] kubernetes daemonset status: `kubectl get ds --all-namespaces`
 - [ ] If a pod/ds is in an error state or pending state `kubectl describe pod -n NAMESPACE POD_NAME`
 - [ ] If a pod/ds is in an error state or pending state `kubectl logs -n NAMESPACE POD_NAME`

 - [ ] Output of running a container on the GPU machine: `docker run -it alpine echo foo`
 - [ ] Docker configuration file: `cat /etc/docker/daemon.json`
 - [ ] Docker runtime configuration: `docker info | grep runtime`

 - [ ] NVIDIA shared directory: `ls -la /run/nvidia`
 - [ ] NVIDIA packages directory: `ls -la /usr/local/nvidia/toolkit`
 - [ ] NVIDIA driver directory: `ls -la /run/nvidia/driver`
 - [ ] kubelet logs `journalctl -u kubelet > kubelet.logs`

**Comment by pichuang** [comment]
Same issue in here

**Comment by creydr** [comment]
same issue here

**Comment by creydr** [comment]
We have the same issue on vanilla kubernetes

- [x] Are you running on an Ubuntu 18.04 node?
- [x]  Are you running Kubernetes v1.13+?
- [x]  Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?
- [ ] Do you have i2c_core and ipmi_msghandler loaded on the nodes?
- only ipmi_msghandler cause of #24 
- [x] Did you apply the CRD (kubectl describe clusterpolicies --all-namespaces)

I our case this leads to unallocable GPUs:

```
$ kubectl describe node xy
...
Capacity:
  ...
  nvidia.com/gpu:     0
Allocatable:
  ...
  nvidia.com/gpu:     0
```

**Comment by pdmack** [comment]
Is this an install on a standalone GPU host with the drivers are already installed? Also, @creydr can you find an existing issue or create an issue for your specific issue?

**Comment by creydr** [comment]
The drivers have not been installed beforehand. @pdmack we have the some issue (getting a `Could not unload NVIDIA driver kernel modules, driver is in use`). The only thing / difference (?!?) is that the capacity & allocateable resources are unset as well.

**Comment by geoberle** [comment]
> Is this an install on a standalone GPU host with the drivers are already installed? Also, @creydr can you find an existing issue or create an issue for your specific issue?

We see the same problem on a fresh RHCOS 4.4 node where no modifications have been made to the operatingsystem whatsoever (no drivers installed prior to the operator installation)

**Comment by shivamerla** [comment]
Its a known issue currently, where driver container always try to unload and install, reload drivers with same version. We will look into this with future releases.

**Comment by maximveksler** [comment]
We are seeing the same issue, on `gpu-operator 1.6.2`

**Comment by lgc0313** [comment]
same error on ocp4.6.19 and gpu-operator 1.6.2

**Comment by jackjliew** [comment]
same error on ocp 4.6.29 and gpu-operator 1.71

**Comment by shivamerla** [comment]
@jackjliew `v1.8.1` has a new initContainer added to driver pod which will handle cleanup of driver modules to avoid this. Let me know if you want to verify with early access bits.

**Comment by shivamerla** [comment]
@mattpielm This has been fixed with `v1.8.x`. Please use `v1.8.2` which was released last week to RH catalog. By the way, you don't need to edit proxy variables manually with driver Daemonset, GPU Operator will automatically inject them if they are added in Cluster-wide proxy.
